packages/@okta/vuepress-site/code/java/spring_security_saml/index.md

```diff
@@ -7,349 +7,375 @@ excerpt: >-
+   import org.springframework.security.saml2.provider.service.authentication.OpenSaml4AuthenticationProvider;
+   import org.springframework.security.saml2.provider.service.authentication.OpenSaml4AuthenticationProvider.ResponseToken;
+   import org.springframework.security.saml2.provider.service.authentication.Saml2AuthenticatedPrincipal;
+   import org.springframework.security.saml2.provider.service.authentication.Saml2Authentication;
+   import org.springframework.security.web.SecurityFilterChain;
+
+   import static org.springframework.security.config.Customizer.withDefaults;
+
+   @Configuration
+   public class SecurityConfiguration {
+
+      @Bean
+      SecurityFilterChain configure(HttpSecurity http) throws Exception {
+
+         OpenSaml4AuthenticationProvider authenticationProvider = new OpenSaml4AuthenticationProvider();
+         authenticationProvider.setResponseAuthenticationConverter(groupsConverter());
+
+         // @formatter:off
+         http
+               .authorizeHttpRequests(authorize -> authorize
+                  .mvcMatchers("/favicon.ico").permitAll()
+                  .anyRequest().authenticated()
+               )
+               .saml2Login(saml2 -> saml2
+                  .authenticationManager(new ProviderManager(authenticationProvider))
+               )
+               .saml2Logout(withDefaults());
+         // @formatter:on
+
+         return http.build();
+      }
+
+      private Converter<OpenSaml4AuthenticationProvider.ResponseToken, Saml2Authentication> groupsConverter() {
+
+         Converter<ResponseToken, Saml2Authentication> delegate =
+               OpenSaml4AuthenticationProvider.createDefaultResponseAuthenticationConverter();
+
+         return (responseToken) -> {
+               Saml2Authentication authentication = delegate.convert(responseToken);
+               Saml2AuthenticatedPrincipal principal = (Saml2AuthenticatedPrincipal) authentication.getPrincipal();
+               List<String> groups = principal.getAttribute("groups");
+               Set<GrantedAuthority> authorities = new HashSet<>();
+               if (groups != null) {
+                  groups.stream().map(SimpleGrantedAuthority::new).forEach(authorities::add);
+               } else {
+                  authorities.addAll(authentication.getAuthorities());
+               }
+               return new Saml2Authentication(principal, authentication.getSaml2Response(), authorities);
+         };
+      }
+   }
+   ```
+
+1. Modify your `build.gradle` file to force the latest version of Open SAML that works with Spring Security 6.
+
+   ```groovy
+   repositories {
+      ...
+      maven { url "https://build.shibboleth.net/nexus/content/repositories/releases/" }
+   }
+
+   dependencies {
+      constraints {
+         implementation "org.opensaml:opensaml-core:4.1.1"
+         implementation "org.opensaml:opensaml-saml-api:4.1.1"
+         implementation "org.opensaml:opensaml-saml-impl:4.1.1"
+      }
+      ...
+   }
```

Comment:
This file was originally indented four spaces.

Reply:
Updated in next commit